userView/service/user.js

```diff
@@ -0,0 +1,52 @@
+const User = require('../models/User');
+const { StatusCodes } = require('http-status-codes');
+const { BadRequestError, NotFoundError, UnAuthenticatedError } = require('../errors/index');
+const checkPermissions = require('../utils/checkPermissions');
+
+
+const viewCurrentUser = async (req, res) => {
+  res.status(StatusCodes.OK).json({ user: req.user });
+};
+
+// update user with user.save()
+const updateUser = async (req, res) => {
+  const { email, name, lastName, location} = req.body;
+  if (!email || !name || !lastName || !location) {
+    throw new BadRequestError('Please provide all values');
+  }
+  const user = await User.findOne({ _id: req.user.userId });
+
+  user.email = email;
+  user.name = name;
+  user.lastName = lastName;
+  user.location = location;
+ 
+  await user.save()
+
+  const token = user.createJWT()
+  res.status(StatusCodes.OK).json({ user, token, location: user.location });
+};
+
+
+const updateUserPassword = async (req, res) => {
+  const { oldPassword, newPassword } = req.body;
+  if (!oldPassword || !newPassword) {
+    throw new BadRequestError('Please provide both values');
+  }
+  const user = await User.findOne({ _id: req.user.userId });
```

Comment:
Everything that has to do with Repository, let it be in one class/ file such that u just import